app/controllers/products_controller.rb

```diff
@@ -0,0 +1,49 @@
+class ProductsController < ApplicationController
```

Comment:
I agree! Controller looks good, thanks for adding the refactoring for `before_action :find_product`! I think this controller as is works for `Admin::ProductsController`.